src/main/java/org/gradle/profiler/mutations/ApplyProjectDependencyChangeSetupMutator.java

```diff
@@ -0,0 +1,66 @@
+package org.gradle.profiler.mutations;
+
+import org.apache.commons.io.FileUtils;
+import org.gradle.profiler.BuildMutator;
+import org.gradle.profiler.ScenarioContext;
+import org.gradle.profiler.mutations.support.FileSupport;
+import org.gradle.profiler.mutations.support.ProjectCombinations;
+
+import java.io.File;
+import java.nio.file.StandardOpenOption;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class ApplyProjectDependencyChangeSetupMutator implements BuildMutator {
+
+    private final String originalText;
+    private final File projectDir;
+    private final File settingsFile;
+    private final ProjectCombinations combinations;
+    private final File generatedProjectsDir;
+
+    protected ApplyProjectDependencyChangeSetupMutator(File projectDir, ProjectCombinations combinations) {
+        this.projectDir = projectDir;
+        this.settingsFile = getSettingsFile();
+        this.originalText = FileSupport.readUnchecked(settingsFile.toPath());
+        this.generatedProjectsDir = new File(projectDir, "gradle-profiler-generated-projects");
+        this.combinations = combinations;
+    }
+
+    private File getSettingsFile() {
+        return Stream.of("settings.gradle", "settings.gradle.kts")
+            .map(it -> new File(projectDir, it))
+            .filter(File::exists)
+            .findFirst()
+            .orElseThrow(() -> new IllegalStateException("No settings.gradle(.kts) file found in " + projectDir));
+    }
+
+    @Override
+    public void beforeScenario(ScenarioContext context) {
+        createProjects(combinations.getProjectNames());
+        String includeProjects = combinations.getProjectNames().stream()
+            .map(it -> {
+                String include = String.format("\ninclude(\"%s\")", it);
+                String projectDir = String.format("\nproject(\":%s\").projectDir = file(\"%s/%s\")", it, generatedProjectsDir.getName(), it);
+                return include + projectDir;
+            })
+            .collect(Collectors.joining());
+        FileSupport.writeUnchecked(settingsFile.toPath(), includeProjects, StandardOpenOption.APPEND);
+    }
+
+    private void createProjects(List<String> projectNames) {
+        for (String projectName : projectNames) {
+            File projectDir = new File(generatedProjectsDir, projectName);
+            projectDir.mkdirs();
+            File buildGradle = new File(projectDir, "build.gradle");
+            FileSupport.writeUnchecked(buildGradle.toPath(), "plugins { id 'java' }");
+        }
+    }
+
+    @Override
+    public void afterScenario(ScenarioContext context) {
+        FileUtils.deleteQuietly(generatedProjectsDir);
+        FileSupport.writeUnchecked(settingsFile.toPath(), originalText);
+    }
```

Comment:
Does this approach work when combining multiple build-script mutators? If it does, cool. If not, then should it? If not, then it should fail in a visible way (i.e. error message preventing multiple build script mutators to be applied).

I think it should work, but for that we need to save the original text in `beforeScenario()`, not in the constructor, no?

Reply:
OK, I see we don't do this in `AbstractFileChangeMutator` either. And if we want to do it, we should call `afterScenario()` in reverse order in `CompositeBuildMutator`... I think we should do this, but not necessarily in this PR.

Reply:
Makes senses, moved logic to the `beforeScenario` for that class and FileChangeMutator.